### PR TITLE
[WIP]`SymmetricGroup` is now a class not function

### PR DESCRIPTION
**Pull request by gxyd**
See commit message for complete "Why this?".

### CODE REVIEW

File: sympy/combinatorics/named_groups.py [hunk 1]
```diff
@@ -267,32 +269,122 @@ def SymmetricGroup(n):
...
+
+
+def _stabilizer(degree, alpha):
+    if degree == 1 or degree == 2:
+        return [Permutation(degree - 1)]
+    gen2_self = Permutation(degree - 1)(0, 1)
+    a = list(range(degree))
+    a.remove(alpha)
+    gen1 = Permutation(*a)
+    if alpha^gen2_self == alpha:
```
Comment (jksuom) [inline sympy/combinatorics/named_groups.py]: Is this condition equivalent to ``alpha > 1``?

File: sympy/combinatorics/named_groups.py [hunk 2]
```diff
@@ -267,32 +269,122 @@ def SymmetricGroup(n):
...
-        G._is_nilpotent = False
-    if n < 5:
-        G._is_solvable = True
-    else:
-        G._is_solvable = False
-    G._degree = n
-    G._is_transitive = True
-    G._is_sym = True
-    return G
+        if alpha == 0:
```
Comment (jksuom) [inline sympy/combinatorics/named_groups.py]: Perhaps ``elif:`` here, too?

**Comment by czgdp1807** [comment]
@jksuom Do we now need to make `SymmetricGroup` a class and not a function? Currently, it is a function.

**Comment by jksuom** [comment]
Permutation groups are instances of `PermutationGroup`. They are defined by giving a list of generating permutations. Each group consists of all permutations obtained by combining the generators.

Symmetric groups are permutation groups that consist of all permutations of a certain degree n. The function `SymmetricGroup` creates a permutation group with a special list of 2 generators. However, there are many other ways to generate a symmetric group. They are all instances of `PermutationGroup` with different generators.

Making `SymmetricGroup` a class implies that its instances would not be instances of `PermutationGroup` and could not be handled by its methods. It seems to me that a function converting a `SymmetricGroup` instance to a `PermutationGroup` instance would be needed. As the current `SymmetricGroup` function already does that, I don't think that it should be changed.

**Comment by czgdp1807** [comment]
Closing.

**Pull request closed without merging**